src/ports/transaction/validation/checkGasPrice.ts

```diff
@@ -92,3 +109,69 @@ const getNetworkGasPrice = async (
 
   return null
 }
+
+/**
+ * Tries to get if the transaction contract method is allowed to skip the max gas price allowed.
+ * It'll return a boolean value
+ * @param components - Config | Contract | Features | Fetcher | Logs components
+ * @param chainId - Network Chain ID
+ * @param transactionData - Transaction data params
+ */
+const isMethodAllowedToSkipMaxGasPrice = async (
+  components: Pick<
+    AppComponents,
+    'config' | 'contracts' | 'features' | 'fetcher' | 'logs'
+  >,
+  transactionData: TransactionData,
+  chainId: ChainId
+) => {
+  const { contracts } = components
+  const manager = getContract(ContractName.CollectionManager, chainId)
+  const factory = getContract(ContractName.CollectionFactoryV3, chainId)
+  const store = getContract(ContractName.CollectionStore, chainId)
+  const collection = getContract(ContractName.ERC721CollectionV2, chainId)
+  const manaConfig = getContract(ContractName.MANAToken, chainId)
+
+  const [contractAddress, fullData] = transactionData['params']
+
+  const { functionSignature: data } = decodeFunctionData(
+    manager.abi, // Either abi works, we just need one that has the executeMetaTransaction method for the first decode
+    'executeMetaTransaction',
+    fullData
+  )
+
+  // Allow the collection manager to create collections using the CollectionFactoryV3
+  if (contractAddress === manager.address) {
+    const { _factory: collectionFactoryAddress } = decodeFunctionData(
+      manager.abi,
+      'createCollection',
+      data
+    )
+
+    return collectionFactoryAddress === factory.address
+  }
+
+  // Approve the collection manager to spend mana on behalf of the user
+  if (contractAddress === manaConfig.address) {
+    const { spender: contractToAuthorizeToSpend } = decodeFunctionData(
+      manaConfig.abi,
+      'approve',
+      data
+    )
```

Comment:
Checking a bit the code, it seems that if the method is not an `approve`, the `decodeFunctionData` can throw:
```
    decodeFunctionData(functionFragment: FunctionFragment | string, data: BytesLike): Result {
        if (typeof(functionFragment) === "string") {
            functionFragment = this.getFunction(functionFragment);
        }

        const bytes = arrayify(data);

        if (hexlify(bytes.slice(0, 4)) !== this.getSighash(functionFragment)) {
            logger.throwArgumentError(`data signature does not match function ${ functionFragment.name }.`,  "data", hexlify(bytes)); <------- throws
        }

        return this._decodeParams(functionFragment.inputs, bytes.slice(4));
    }
```
Does this code allow calling other functions of the contracts?

Reply:
good catch! this code will throw an error if the method sent is not an `approve`. same for the other validations. I'll wrap it into a `try/catch` to return `false`